packages/sail_ui/lib/widgets/inputs/dropdown_menu.dart

```diff
@@ -0,0 +1,177 @@
+      );
+    }
+
+    return SizedBox(
+      width: widget.width,
+      height: widget.large ? 32 : 24,
+      child: _Button(
+        large: widget.large,
+        padding: EdgeInsets.only(
+          left: 8,
+          right: widget.icon == null ? 8 : 4,
+        ),
+        onPressed: () {
+          var bounds = getGlobalBoundsForContext(context);
+          showSailMenu(
+            context: context,
+            preferredAnchorPoint: Offset(
+              bounds.left - (context.isWindows ? 1 : 9),
+              bounds.top - offsetY - 3,
+            ),
+            menu: SailMenu(
+              items: items,
+              width: widget.width,
+            ),
+          );
+        },
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          mainAxisSize: MainAxisSize.max,
+          children: [
+            currentItem,
+            if (widget.icon != null)
+              Padding(
+                padding: const EdgeInsets.only(left: 4),
+                child: widget.icon,
+              ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class SailDropdownItem<T> extends StatelessWidget {
+  const SailDropdownItem({
+    required this.value,
+    required this.child,
+    super.key,
+  });
+
+  final T value;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return Align(
+      alignment: Alignment.centerLeft,
+      child: child,
+    );
+  }
+}
+
+class _Button extends StatelessWidget {
+  final VoidCallback? onPressed;
+  final Widget child;
+  final EdgeInsets padding;
+  final bool large;
+
+  const _Button({
+    required this.onPressed,
+    required this.child,
+    this.padding = const EdgeInsets.symmetric(
+      horizontal: 12,
+      vertical: 0,
+    ),
+    required this.large,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+
+
+    Color textColor;
+    final sailTheme = context.sailTheme;
+    if (onPressed != null) {
+      textColor = sailTheme.colors.text;
+    } else {
+      textColor = sailTheme.colors.textTertiary;
+    }
+
+    return SizedBox(
+      height: large ? 32 : 24,
+      child: MaterialButton(
+        // textTheme: textTheme,
+        color: sailTheme.colors.background,
+        shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.all(Radius.circular(context.isWindows ? 3 : 4)),
+          side: BorderSide(
+            color: sailTheme.colors.divider,
+            width: 0.5,
+          ),
+        ),
+        onPressed: onPressed,
+        elevation: 0,
+        hoverElevation: 0,
+        focusElevation: 0,
+        minWidth: 32,
+        padding: EdgeInsets.zero,
+        child: Container(
+          alignment: Alignment.center,
+          height: large ? 32 : 24,
+          padding: padding,
+          child: DefaultTextStyle(
+            style: SailStyleValues.twelve.copyWith(
+              color: textColor,
+            ),
+            child: child,
+          ),
+        ),
```

Comment:
I know it's a common flutter-pattern, but I don't like it. If I send in a child where I want to set it to text size 15, it's being overridden here. If it takes in a widget, the parent should decide what that widget should look like imo.

Reply:
`DefaultTextStyle` only provides a default akin to `TextTheme` so if you specify a specific style on the child, it takes precedence over the `DefaultTextStyle` values.

Still, if we don't want that behavior anyway I'll just remove it!